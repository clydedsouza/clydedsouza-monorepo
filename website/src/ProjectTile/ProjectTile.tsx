export interface IProjectTileProps {
  name: string
  image: string
  description: string
}

function ProjectTile(props: IProjectTileProps) {
  const { name, image, description } = props
  return (
    <div>
      <p>Project</p>
      <p>ifbwebfwebfi</p>
    </div>
  )
}

export default ProjectTile
