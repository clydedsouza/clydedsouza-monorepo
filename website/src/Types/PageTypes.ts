export enum PageTypes {
  About = 'About',
  Projects = 'Projects',
  Speaking = 'Speaking',
  Teaching = 'Teaching',
  Books = 'Books',
  Highlights = 'Highlights',
  Pinned = 'Pinned',
}
