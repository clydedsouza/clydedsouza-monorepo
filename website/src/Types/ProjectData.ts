export interface IProjectData {
  data: IProject[]
}

export interface IProject {
  id: string
  title: string
  date: string
  description: string
  hasWebsite: boolean
  website: string
  onGithub: boolean
  github: string
  madeUsing: string[]
  category: string
  image: string
  imageDescription: string
}
