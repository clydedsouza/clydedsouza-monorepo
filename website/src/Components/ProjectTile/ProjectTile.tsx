import { IProject } from '../../Types/ProjectData'
import Links from './Links/Links'
import MadeUsing from './MadeUsing/MadeUsing'
import './ProjectTile.scss'

function ProjectTile(props: IProject) {
  return (
    <div className="project-tile">
      <div
        style={{
          backgroundImage: `url(${props.image}), url("https://files.clydedsouza.net/images/projects/omg-team.jpg")`,
        }}
      >
        <span>{props.category}</span>
      </div>
      <div>
        <h3>{props.title}</h3>
        <p>{props.description}</p>
        <MadeUsing {...props} />
        <Links {...props} />
      </div>
    </div>
  )
}

export default ProjectTile
