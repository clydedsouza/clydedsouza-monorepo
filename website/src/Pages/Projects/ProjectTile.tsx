import { IProject } from '../../Types/ProjectData'

function ProjectTile(props: IProject) {
  return (
    <section>
      <div
        className="project-header"
        style={{ backgroundImage: `url(${props.image})` }}
      >
        <div className="project-header-contents">
          <div>
            <span>{props.category}</span>
            {!props.isActive && <span>Inactive</span>}
          </div>
        </div>
      </div>
      <div className="project-body">
        <h3>{props.title}</h3>
        <p>{props.description}</p>
        <p>
          {props.madeUsing &&
            props.madeUsing.map((item) => <span key={item}>{item}</span>)}
        </p>
        <div>
          <div className="col-xs-8 text-left">&nbsp;</div>
          <div className="col-xs-4 text-right">
            {props.isActive && props.hasWebsite && (
              <a href={props.website} target="_blank" rel="noreferrer">
                <i className="fas fa-globe"></i>
              </a>
            )}
            {props.onGithub && (
              <a href={props.github} target="_blank" rel="noreferrer">
                <i className="fab fa-github"></i>
              </a>
            )}
          </div>
        </div>
      </div>
    </section>
  )
}

export default ProjectTile
