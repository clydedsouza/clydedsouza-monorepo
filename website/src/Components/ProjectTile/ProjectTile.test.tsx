import { render } from '@testing-library/react'
import { IProject } from '../../Types/ProjectData'
import ProjectTile from './ProjectTile'

const emptyProject: IProject = {
  title: '',
  description: '',
  image: '',
  imageDescription: '',
  category: '',
  date: '',
  id: '',
  madeUsing: [],
  hasWebsite: true,
  website: 'test',
  onGithub: true,
  github: 'test',
}

const projectWithoutLinks: IProject = {
  id: '20121201-tulips-cover',
  title: 'Tulips cover page designs',
  date: '01 Dec 2012',
  description:
    'I was responsible for designing the cover page for Tulips - a church bulletin. Each cover page was designed in accordance to a theme for that particular issue and conveyed the underlying meaning of the theme.',
  hasWebsite: false,
  website: '',
  onGithub: false,
  github: '',
  madeUsing: ['Photoshop'],
  category: 'Graphic Design',
  image: 'https://files.clydedsouza.net/images/projects/tulips-collage.jpg',
  imageDescription: "Church magazine cover designs by Clyde D'Souza",
}

const projectWithoutDescription: IProject = {
  id: '20181024-chemelonj',
  title: 'Chameleon',
  date: '24 Oct 2018',
  description: '',
  hasWebsite: true,
  website: 'https://chemeleon.net',
  onGithub: true,
  github: 'https://github.com/chemeleon/chemeleon.github.io/',
  madeUsing: ['html', 'sass'],
  category: 'Website',
  image: 'https://files.clydedsouza.net/images/projects/chemeleon.jpg',
  imageDescription: 'test',
}

const projectWithAllInfo: IProject = {
  id: '20170701-profile-sticker',

  title: 'Profile Sticker',
  date: '01 Jul 2017',
  description:
    'Profile Sticker is a web app that allows a user to add a sticker from different themes to their profile picture. The user can choose to upload a picture from and download to a computer or Facebook.',
  hasWebsite: true,
  website: 'https://profilesticker.net',
  onGithub: true,
  github: 'https://github.com/profilesticker/profilesticker.github.io/',
  madeUsing: ['HTML', 'CSS', 'SCSS', 'JavaScript', 'Mustache.js'],
  category: 'Website',
  image:
    'https://files.clydedsouza.net/images/projects/profilesticker-website.png',
  imageDescription:
    'Add a sticker to your profile picture using Profile Sticker in just 4 easy steps.',
}

describe('Project tile', () => {
  it('should render tile', () => {
    render(<ProjectTile {...projectWithAllInfo} />)
    expect(document.body).toMatchSnapshot()
  })

  it('should render tile with empty project', () => {
    render(<ProjectTile {...emptyProject} />)
    expect(document.body).toMatchSnapshot()
  })

  it('should render tile without links', () => {
    render(<ProjectTile {...projectWithoutLinks} />)
    expect(document.body).toMatchSnapshot()
  })

  it('should render tile without description', () => {
    render(<ProjectTile {...projectWithoutDescription} />)
    expect(document.body).toMatchSnapshot()
  })
})
