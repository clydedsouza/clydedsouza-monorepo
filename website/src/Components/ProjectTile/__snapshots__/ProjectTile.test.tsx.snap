// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project tile should render tile 1`] = `
<body>
  <div>
    <div
      class="project-tile"
    >
      <div>
        <span>
          Website
        </span>
      </div>
      <div>
        <h3>
          Profile Sticker
        </h3>
        <p>
          Profile Sticker is a web app that allows a user to add a sticker from different themes to their profile picture. The user can choose to upload a picture from and download to a computer or Facebook.
        </p>
        <div
          class="made-using"
        >
          <span>
            HTML
          </span>
          <span>
            CSS
          </span>
          <span>
            SCSS
          </span>
          <span>
            JavaScript
          </span>
          <span>
            Mustache.js
          </span>
        </div>
        <div
          class="project-links"
        >
          <a
            href="https://profilesticker.net"
            rel="noreferrer"
            target="_blank"
          >
            <i
              class="fas fa-globe"
            />
          </a>
          <a
            href="https://github.com/profilesticker/profilesticker.github.io/"
            rel="noreferrer"
            target="_blank"
          >
            <i
              class="fab fa-github"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Project tile should render tile with empty project 1`] = `
<body>
  <div>
    <div
      class="project-tile"
    >
      <div>
        <span />
      </div>
      <div>
        <h3 />
        <p />
        <div
          class="project-links"
        >
          <a
            href="test"
            rel="noreferrer"
            target="_blank"
          >
            <i
              class="fas fa-globe"
            />
          </a>
          <a
            href="test"
            rel="noreferrer"
            target="_blank"
          >
            <i
              class="fab fa-github"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Project tile should render tile without description 1`] = `
<body>
  <div>
    <div
      class="project-tile"
    >
      <div>
        <span>
          Website
        </span>
      </div>
      <div>
        <h3>
          Chameleon
        </h3>
        <p />
        <div
          class="made-using"
        >
          <span>
            html
          </span>
          <span>
            sass
          </span>
        </div>
        <div
          class="project-links"
        >
          <a
            href="https://chemeleon.net"
            rel="noreferrer"
            target="_blank"
          >
            <i
              class="fas fa-globe"
            />
          </a>
          <a
            href="https://github.com/chemeleon/chemeleon.github.io/"
            rel="noreferrer"
            target="_blank"
          >
            <i
              class="fab fa-github"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Project tile should render tile without links 1`] = `
<body>
  <div>
    <div
      class="project-tile"
    >
      <div>
        <span>
          Graphic Design
        </span>
      </div>
      <div>
        <h3>
          Tulips cover page designs
        </h3>
        <p>
          I was responsible for designing the cover page for Tulips - a church bulletin. Each cover page was designed in accordance to a theme for that particular issue and conveyed the underlying meaning of the theme.
        </p>
        <div
          class="made-using"
        >
          <span>
            Photoshop
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
